### ISSUE TITLE
[Sybil Report]

### ISSUE THREAD

**Issue by shsh-sh** [LayerZero-Labs/sybil-report#864]
\n---\nname: Sybil Report\nabout: Report Sybil Activity on LayerZero\ntitle: \"[Sybil Report]\"\nlabels: under review\nassignees: LayerZero-GH\n\n---\n\n# Reported Addresses<br>\n\n0x98f7212b78294b8345359ec28ad02ef5e14cdd07<br>0x5a41b9bb8fe3e422d443ab75519406e016064c11<br>0xe5f19c1c0cd12f26242fbbb0d5cb96226cfeed10<br>0x7c56dcb86d395101b5ecade0464b22583780102b<br>0x12c09a4821ad4c874980c5249ff98e15c5a636d8<br>0xf19a68e6d02532601754955736bdc7460e2a6bc4<br>0x387ad6ea33bf4c03ec91befd8e6b1ac1de89ec5d<br>0x3db9af49332976f051d1a60df90a1fcb1dcf014c<br>0x3b9e6268e952cf26d0621d49839051deeeb613cf<br>0x8ea0fedc35aee19d1de1474ff5d26b53f8ad2453<br>0x4c187bea5a7444b2f6a5ca0c3d9837b44e9f56b5<br>0xad5009a4336a9fe8bf22821a8e1cd9bccbcb4291<br>0x1be83977b91fe28f0c8dc7ba974a8efc4e7aa986<br>0xa4b2781f75184624c30a52d189d7ac8b628a1fd9<br>0xf83c510b2f559f58ed83099f40908907c62317ff<br>0x81dac2f6acd25192c83ce74ac8ad8c83eef36f16<br>0x9fd685f172849c6586249a4fa068eed89e54ebc3<br>0x1b391aa9c388ac6d69d4e5a90e41a568308116cf<br>0x2bafbd0d27dd756c89595e90b6a88dd05b9d0915<br>0x7fa5ed63840b20999b27b94a396d9633d2262c46<br>0xeb06069c62ada4cbd750c86908ad180d3adc25ac\n\n# Description<br>\n\nThis cluster contains 21 addresses, each of which corresponds to all provided statistics and can be easily checked manually at https://layerzeroscan.com/.<br><br>\nAll addresses in the cluster use the same networks: [Avalanche, BNB Chain, Optimism, Polygon].<br>Cluster-averaged correlation of vectors of used networks: 0.9100942116586188.<br><br>Cluster median correlation of the vectors of used networks: 0.9283443892592866.<br>\nAll addresses in the cluster use the same protocols: [Aptos Bridge, Bitcoin Bridge, Stargate].<br>Cluster-averaged correlation of vectors of used protocols: 1.0.\n<br>Cluster median correlation of the vectors of used protocols: 1.0.\n<br>\nAll addresses in the cluster have from 1111115.81 to 1112040.333 (avg 1111370.222) seconds interval between transactions within the cluster.<br>Similarity rate across the cluster by interval between transactions: 1.0<br><br>\nAll addresses in the cluster have 22 transactions count.<br>Similarity rate across the cluster by number of transactions: 1.0.<br><br>\nAll addresses in the cluster have 22 unique* transactions count.<br>Similarity rate across the cluster by number of unique* transactions: 1.0.<br><br>\nAll addresses in the cluster have 0.952 share of transactions with sequentially* used networks.<br>Similarity rate across the cluster by share of transactions with sequentially* used networks: 1.0.<br><br>\nAll cluster addresses share a great deal of similarity in other ways that can be found at https://debank.com/profile/.<br>All addresses in the cluster have from 1687873419 to 1687874176 (avg 1687873793.143) born timestamp of the first transaction among all blockchains.<br>Similarity rate across the cluster by accounts born timestamp: 1.0.<br><br>\n\nAll of the above statistics confirm that the clustering was done correctly and all attached addresses are Sybils. Address transactions are performed on the same days, at approximately the same time, using the same protocols and networks. The transactions were probably performed by a script or sequentially manually on the accounts.<br><br>\n\n# Detailed Methodology & Walkthrough<br>\n\n1. data processing:<br>\n1.1 Removing duplicates.<br>\n1.2 Filling empty values.<br>\n1.3 Generation of user and transaction datasets.<br>\n2. Collecting and analyzing statistics:<br>\n2.1. Obtaining mean and median values for number of transactions, number of unique* transactions, interval between transactions, networks and protocols used, transactions with sequentially* used networks.<br>\n2.2 First attempts at clustering, selection of the best criteria and normalization of the data. Data reflecting one criterion and not paired with other criteria were not normalized, and other data were presented from 0 to 1.<br>\n3. Multistage clustering. For clustering, model sklearn.cluster.DBSCAN was mainly used because it does not require specifying the number of clusters, can be executed on multiple CPU cores at once, and has flexible tuning.<br>\n3.1 Clustering by protocols and networks, and then forming subclusters by transactions. For clustering by networks, a vector was constructed for each address to represent the number of times the user used each network. For further clustering by transactions, a vector was generated for each address reflecting the number of user transactions for every 4 hours (this proved to be the best time to optimize the computation and not degrade the clustering results). Clustering was performed with parameters `eps=2`, `min_samples=20`.<br>\n3.2 Clustering by normalized statistics. Clustering on normalized data was performed with parameters `eps=0.3`, `min_samples=20` or `eps=0.6`.<br>\n3.3 Clustering using additional metrics taken from debank. For example, account birth timestamp, all networks used by the account and also account balances in networks. The timestamp was used to select accounts created around the same time, for which further clustering by transaction similarity was performed. `eps=4` and `min_samples=20` was used.<br>\n3.4 Searching for cluster intersections. The search for cluster intersections was performed using set() & set() to search for common addresses.<br>\n3.5 Plotting. For each address, a graph was plotted to visualize the relationship with other addresses in the cluster.<br>\n3.6 Manual address checks, checks via https://layerzeroscan.com/ and crosschecking crosschain statistics via https://debank.com/profile/.<br>\n\nThe code snippets used in the clustering are here: https://github.com/shsh-sh/hlibys.<br>\nMost of the work was done by doing a long and deep analysis, brute force search of the best criteria, manual cluster checking and making sure that the clusters found contained only Sybil addresses.<br>\n\nNormalizing the data, constructing correct vectors, partitioning into clusters and subclusters in two different ways and further searching for address overlaps in subclusters and checking against statistical data can ensure that the submitted addresses are Sybils driven.<br>\n\nExplanation:<br>\n* Unique transactions - transactions of one address with unique `SOURCE_TRANSACTION_HASH`. Used to ignore transactions that were made through one protocol to many networks simultaneously.<br>\n* Transactions with sequentially used networks - successive transactions in which `SOURCE_CHAIN` is equal to `DESTINATION_CHAIN` of the previous transaction.<br>\n\n# Reward Address (If Eligible)<br>\n0x7DAB82514de70c45B459E884EBe57Ac3d91eeE09<br>\n
